ratelimit/ratelimit.go

```diff
@@ -3,8 +3,8 @@ package ratelimit
 import (
 	"bytes"
 	"context"
-	"crypto/sha256"
-	"encoding/hex"
+	"crypto/md5" // #nosec
```

Comment:
since we're here, do we even need a crypto level hash for this? Some time ago, I found a small analysis for another use case, and to me it seemed that FNV is quite OK for keys, when security is not in concern. (Cannot find the exact reference, but there are similar results published.)

Reply:
If we want to secure the data, because we likely store authorization headers in redis I am more in favor of a strong secure hash.

Reply:
I think this is not the reason why crypto hashes need to be strong. We never compare these particular hashes in any real security scenario.

Reply:
ok, I think I see the reason